### DISCUSSION TITLE
Вопрос по установке корневых сертификатов

### DISCUSSION THREAD

**Discussion by Poisent** [CryptoPro/libcore#15] [Q&A] [answered]
Добрый день! Коллеги, возник вопрос по установке корневых сертификатов в uRoot.
Можно ли как-то отключить ввод подтверждения? Из-за требования подтверждения от пользователя, не получается устанавливать сертификаты программным путём. 
При установке корневых сертификатов в отличные от uRoot хранилища, корневые сертификаты при проверке не определяются.
Есть ли какое-то решение для программной установки корневых сертификатов? 
Устанавливаем след образом:
 using var store = new CpX509Store(StoreName.Root, StoreLocation.CurrentUser);
 store.Open(OpenFlags.MaxAllowed);
store.Add(cert);

**Comment by maxdm** [comment, marked as answer]
Отключить подтверждение нельзя, установка корневых сертификатов очень важное действие с точки зрения безопасности.
Но можно установить в mRoot, если есть права root

26 апр. 2023 г., в 16:53, Poisent ***@***.***> написал(а):

﻿

Добрый день! Коллеги, возник вопрос по установке корневых сертификатов в uRoot.
Можно ли как-то отключить ввод подтверждения? Из-за требования подтверждения от пользователя, не получается устанавливать сертификаты программным путём.
При установке корневых сертификатов в отличные от uRoot хранилища, корневые сертификаты при проверке не определяются.
Есть ли какое-то решение для программной установки корневых сертификатов?

**Reply by Poisent** [reply]
при попытке установить сертификаты в mRoot получаем вот такую ошибку:
Provider DLL failed to initialize correctly.
Ошибка одинаковая, через sudo и без. Через приложение и утилиту certmgr. В чем может быть дело?

**Comment by maxdm** [comment]
Очень странно. Загляните в syslog
certmgr -list -store uRoot работает?

26 апр. 2023 г., в 19:18, Poisent ***@***.***> написал(а):

﻿

при попытке установить сертификаты в mRoot получаем вот такую ошибку:
Provider DLL failed to initialize correctly.
Ошибка одинаковая, через sudo и без. Через приложение и утилиту. В чем может быть дело?

**Comment by maxdm** [comment]
-mRoot конечно же

26 апр. 2023 г., в 19:50, Коллегин Максим ***@***.***> написал(а):

﻿
Очень странно. Загляните в syslog
certmgr -list -store uRoot работает?

26 апр. 2023 г., в 19:18, Poisent ***@***.***> написал(а):

﻿

при попытке установить сертификаты в mRoot получаем вот такую ошибку:
Provider DLL failed to initialize correctly.
Ошибка одинаковая, через sudo и без. Через приложение и утилиту. В чем может быть дело?

**Reply by Poisent** [reply]
certmgr -list -store mRoot не работает.
В командной строке ошибка: Provider DLL failed to initialize correctly
В системном логе:
<capi20>CertOpenStore!failed: LastError = 0x8009001D
Нет мыслей, в чем может быть дело?

**Reply by Poisent** [reply]
Проблема решилась переустановкой крипто про
